lambda/js/src/main/scala/feral/lambda/IOLambdaPlatform.scala

```diff
@@ -34,10 +36,15 @@ private[lambda] trait IOLambdaPlatform[Event, Result] {
   private lazy val handlerFn
       : js.Function2[js.Any, facade.Context, js.Promise[js.Any | Unit]] = {
     (event: js.Any, context: facade.Context) =>
-      (for {
-        lambda <- setupMemo
-        event <- IO.fromEither(decodeJs[Event](event))
-        result <- lambda(event, Context.fromJS(context))
-      } yield result.map(_.asJsAny).orUndefined).unsafeToPromise()(runtime)
+      setupMemo.toJSPromise(scala.concurrent.ExecutionContext.parasitic).`then` {
```

Comment:
Try this, I think it's just type inference problems :)

```suggestion
      setupMemo.toJSPromise(scala.concurrent.ExecutionContext.parasitic).`then`[js.Any] {
```